pkg/db/v1beta1/postgres/postgres.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright 2022 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package postgres
+
+import (
+	crand "crypto/rand"
+	"database/sql"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"os"
+	"time"
+
+	_ "github.com/lib/pq"
+
+	v1beta1 "github.com/kubeflow/katib/pkg/apis/manager/v1beta1"
+	"github.com/kubeflow/katib/pkg/db/v1beta1/common"
+	"github.com/kubeflow/katib/pkg/util/v1beta1/env"
+	"k8s.io/klog"
+)
+
+const (
+	dbDriver = "postgres"
+
+	connectInterval = 5 * time.Second
+	connectTimeout  = 60 * time.Second
+)
+
+type dbConn struct {
+	db *sql.DB
+}
+
+func getDbName() string {
+	dbPassEnvName := common.DBPasswordEnvName
+	dbPass := os.Getenv(dbPassEnvName)
+
+	dbUser := env.GetEnvOrDefault(
+		common.DBUserEnvName, common.DefaultPostgreSQLUser)
+	dbHost := env.GetEnvOrDefault(
+		common.PostgreSQLDBHostEnvName, common.DefaultPostgreSQLHost)
+	dbPort := env.GetEnvOrDefault(
+		common.PostgreSQLDBPortEnvName, common.DefaultPostgreSQLPort)
+	dbName := env.GetEnvOrDefault(common.DefaultPostgreSQLDatabase,
+		common.DefaultPostgreSQLDatabase)
+
+	psqlInfo := fmt.Sprintf("host=%s port=%s user=%s "+
+		"password=%s dbname=%s sslmode=disable",
+		dbHost, dbPort, dbUser, dbPass, dbName)
+
+	return psqlInfo
+}
+
+func NewDBInterface() (common.KatibDBInterface, error) {
+	db, err := openSQLConn(dbDriver, getDbName(), connectInterval, connectTimeout)
+	if err != nil {
+		return nil, fmt.Errorf("DB open failed: %v", err)
+	}
+	return NewWithSQLConn(db)
+}
+
+func openSQLConn(driverName string, dataSourceName string, interval time.Duration,
```

Comment:
Can we move `openSQLConn` function to the common package to make it common?

Reply:
Sure :)

Reply:
DONE :) Thanks for removing duplicates

Reply:
Thanks for updating!